scalameta/contrib/src/main/scala/scala/meta/contrib/instances/EqualityInstances.scala

```diff
@@ -1,26 +1,33 @@
-package scala.meta.contrib.equality
+package scala.meta.contrib.instances
+
+import scala.meta._
+import scala.meta.contrib.Equal
+import scala.meta.prettyprinters.{Show, Structure}
+import scala.language.higherKinds
+
+trait EqualityInstances {
+  implicit def showEquality[F[x] <: Show[x]]: Equal[F] =
+    new Equal[F] {
+      override def isEqual[A](a: A, b: A): Boolean =
+        a.show[F].equals(b.show[F])
```

Comment:
I suspect the performance penalty of using Show[T] is quite high. For large trees, `.structure` builds MASSIVE strings for the entire tree while Structurally allocates almost nothing and fails fast on the first mismatching tree node.

We'll need benchmarks to confirm how significant the difference is. I was working last night on adding JMH benchmarks to scala.meta, following up from a hacking session with @rorygraves trying to speed up parsing/tree traversals.